include/tvm/runtime/packed_func.h

```diff
@@ -371,6 +371,20 @@ class TVMPODValue_ {
     TVM_CHECK_TYPE_CODE(type_code_, kDLFloat);
     return value_.v_float64;
   }
+  operator float() const {
+    double tmp_v = 0;
+    if (type_code_ == kDLInt) {
+      tmp_v = static_cast<double>(value_.v_int64);
+      CHECK_LE(tmp_v, std::numeric_limits<float>::max());
```

Comment:
this three lines are redundent with the below 3 lines. can you have if else that set tmp_v's value?

Reply:
Depends whether we want to run TYPE_CHECK. If we want to run type check we have to have these redundant code.